client/src/components/AppMenu/AppMenu.js

```diff
@@ -84,7 +84,7 @@ function AppMenu() {
       tabIcon: <ShoppingCartOutlinedIcon />,
       tabPrimaryText: 'Cart',
       onClick: dispatchCloseMenu,
-      linkTo: '/card'
+      linkTo: '/cart'
```

Comment:
The file is giantic, you should split better, in term of react, event text, button, input, all should be separate components.

MenuInterface - use camel case for naming. Could be declared outside of the component to make is smaller.

currencyItems also should be constant declared outside of the component

Reply:
It's better to avoid such long relative paths, 

import { switchCurrency } from '../../redux/reducers/appCurrencyReducer';

Read about aliases in webpack.

Reply:
onClick={[openTab, tab.onClick]} It's bad, you create arr on the fly, memo for this component AppMenuTab will not work.